observations/forward_operators/obs_def_abs_humidity_mod.f90

```diff
@@ -0,0 +1,186 @@
+
+use ensemble_manager_mod,  only : ensemble_type
+use obs_def_utilities_mod, only : track_status
+
+
+implicit none
+private
+
+public :: get_expected_absolute_humidity
+
+! version controlled file description for error handling, do not edit
+character(len=256), parameter :: source   = &
+   "$URL$"
+character(len=32 ), parameter :: revision = "$Revision$"
+character(len=128), parameter :: revdate  = "$Date$"
+
+logical, save       :: module_initialized   = .false.
+logical, save       :: first_time_warn_low  = .true.
+logical, save       :: first_time_warn_high = .true.
+character(len=512)  :: msgstring
+real(r8), parameter :: MIN_VALUE = 0.0
+real(r8), parameter :: MAX_VALUE = 0.1
+
+contains
+
+!----------------------------------------------------------------------
+
+subroutine initialize_module
+
+call register_module(source, revision, revdate)
+module_initialized = .true.
+
+end subroutine initialize_module
+
+!----------------------------------------------------------------------------
+
+subroutine get_expected_absolute_humidity(state_handle, ens_size, location, ah, istatus)
+
+type(ensemble_type), intent(in)     :: state_handle
+integer,             intent(in)     :: ens_size
+type(location_type), intent(in)     :: location
+real(r8),            intent(out)    :: ah(ens_size)    ! absolute humidity
+integer,             intent(out)    :: istatus(ens_size)
+
+real(r8), dimension(ens_size) :: qvap, tmpk, pres, rho_dry, tmpv
+integer,  dimension(ens_size) :: qvap_istatus, tmpk_istatus, pres_istatus
+real(r8)                      :: xyz(3)
+logical :: return_now
+
+if ( .not. module_initialized ) call initialize_module
+
+istatus = 0
+
+!  interpolate the mixing ratio to the location
+call interpolate(state_handle, ens_size, location, QTY_VAPOR_MIXING_RATIO, qvap, qvap_istatus)
+call track_status(ens_size, qvap_istatus, ah, istatus, return_now)
+
+where (istatus == 0 .and. qvap < 0.0_r8) qvap = epsilon(0.0_r8)  ! clamping?
+where (istatus /= 0) istatus = 99
+if(return_now) return
+
+!  interpolate the temperature to the desired location
+call interpolate(state_handle, ens_size, location, QTY_TEMPERATURE, tmpk, tmpk_istatus)
+call track_status(ens_size, tmpk_istatus, ah, istatus, return_now)
+
+where (tmpk <= 0.0_r8)
+   ah = missing_r8
+   istatus = 99
+endwhere
+
+if(all(istatus /= 0)) return  ! not using return_now because where clause modifies istatus.
+
+!  interpolate the pressure, if observation location is not pressure
+if ( is_vertical(location, "PRESSURE") ) then
+   xyz = get_location(location)
+   pres = xyz(3)
+else
+   ! pressure comes back in pascals (not hPa or mb)
+   call interpolate(state_handle, ens_size, location, QTY_PRESSURE, pres, pres_istatus)
+   call track_status(ens_size, pres_istatus, ah, istatus, return_now)
+
+   where (pres <= 0.0_r8 .or. pres >= 120000.0_r8)
+      ah = missing_r8
+      istatus = 99
+   end where
+   if(all(istatus /= 0)) return  ! not using return_now because where clause modifies istatus.
```

Comment:
the trailing comment should be removed - 'return_now()' is not an issue.

Reply:
i'm not sure about removing this comment.  generally we tell people to call track_status() and then test the logical 'return_now' to decide whether all the forward operators have failed and it's time to return.  in this case, return_now is not consistent anymore with the istatus codes if more have been set to fail.  this might be a useful comment - but maybe in fact it needs more information if it didn't help tim.

Reply:
I am removing the comment for now. These comments are inherited from obs_def_rel_humidity_mod.f90 from which I copied.

Reply:
That's what I thought.